glib/src/variant.rs

```diff
@@ -244,15 +244,27 @@ impl Variant {
     ///
     /// * if `self` is not a container type.
     /// * if given `index` is larger than number of children.
-    #[doc(alias = "get_child_value")]
-    #[doc(alias = "g_variant_get_child_value")]
     pub fn child_value(&self, index: usize) -> Variant {
-        assert!(index < self.n_children());
         assert!(self.is_container());
+        assert!(index < self.n_children());
 
         unsafe { from_glib_full(ffi::g_variant_get_child_value(self.to_glib_none().0, index)) }
     }
 
+    /// Try to read a child item out of a container `Variant` instance.
+    ///
+    /// It returns `None` if `self` is not a container type or if the given
+    /// `index` is larger than number of children.
+    #[doc(alias = "g_variant_get_child_value")]
+    pub fn get_child_value(&self, index: usize) -> Option<Variant> {
```

Comment:
Would call the method try_child_value maybe ?

Reply:
Hmm, I'd probably say no, naming pattern is already similar to other index-based getters in the stdlib like https://doc.rust-lang.org/std/primitive.slice.html#method.get.

Reply:
I think the naming is a bit strange now. `child_value()` and `get_child_value()` sound like they would do the same, but they actually don't.

As such I would also call this `try_child_value()`

Reply:
Ack, renamed that way.